metadata-ingestion/src/datahub/ingestion/source/snowflake/snowflake_lineage_v2.py

```diff
@@ -109,19 +109,31 @@ def get_workunits(
             f"Upstream lineage detected for {self.report.num_tables_with_upstreams} tables.",
         )
 
-        yield from self.get_table_external_upstream_workunits()
+        if self._external_lineage_map:  # Some external lineage is yet to be emitted
```

Comment:
Do we need this if statement at all? I see it was in the old code, but this is initialized to a `defaultdict` so I don't think it should ever be `None`

Reply:
This if here is added to check whether self._external_lineage_map is nonempty.